salt/metalk8s/macro.sls

```diff
@@ -2,12 +2,16 @@
 
 {%- macro pkg_installed(name='') -%}
   {%- set package = repo.packages[name] | default({}) %}
+  {%- set version = package.version | default(None) %}
+  {%- set deps_list = salt.metalk8s_package_manager.list_pkg_deps(name, version) %}
+  {%- for name, version in deps_list.items() %}
```

Comment:
Also, doing the for here will never work. Because you will create one `pkg.installed` per deps it's not what we want. We want to give all deps in one command so that yum can use them to one `yum` command